hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,679 @@ def _estimate_elbow_position(self, curve_values):
         # more or less defines a triangle
         # The elbow should be the point which is the
         # furthest distance from this line
-
-        y2 = np.log(curve_values[maxpoints])
+        if log:
+            y2 = np.log(curve_values[maxpoints])
+            y1 = np.log(curve_values[0])
+        else:
+            y2 = curve_values[maxpoints]
+            y1 = curve_values[0]
         x2 = maxpoints
-        y1 = np.log(curve_values[0])
         x1 = 0
         # loop through the curve values and calculate
         distance = np.zeros(maxpoints)
         for i in range(maxpoints):
-            y0 = np.log(curve_values[i])
+            #y0 = np.log(curve_values[i]+1)
```

Comment:
where has the +1 come from and why is it commented out?

Reply:
leftover code... log can sometimes throw errors depending on value so played with a shift of 1 to ensure this doesn't occur.

Reply:
Now Removed